pkg/workers/leader_election_mgr_test.go

```diff
@@ -185,3 +187,385 @@ func TestLeaderElectionManager_acquireLeaderLease(t *testing.T) {
 		})
 	}
 }
+
+func TestNewLeaderElectionManager(t *testing.T) {
+	type args struct {
+		workers           []Worker
+		connectionFactory *db.ConnectionFactory
+		reconcilerConfig  *ReconcilerConfig
+	}
+	tests := []struct {
+		name string
+		args args
+		want *LeaderElectionManager
+	}{
+		{
+			name: "should return a New Leader Election Manager",
+			args: args{
+				workers:           []Worker{},
+				connectionFactory: &db.ConnectionFactory{},
+				reconcilerConfig: &ReconcilerConfig{
+					ReconcilerRepeatInterval:               30 * time.Second,
+					LeaderElectionReconcilerRepeatInterval: 15 * time.Second,
+					LeaderLeaseExpirationTime:              1 * time.Minute,
+				},
+			},
+			want: &LeaderElectionManager{
+				workers:                                []Worker{},
+				connectionFactory:                      &db.ConnectionFactory{},
+				tearDown:                               nil,
+				leaderElectionReconcilerRepeatInterval: 15 * time.Second,
+				leaderLeaseExpirationTime:              1 * time.Minute,
+				workerGrp:                              sync.WaitGroup{},
+			},
+		},
+	}
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			t.Parallel()
+			g.Expect(NewLeaderElectionManager(tt.args.workers, tt.args.connectionFactory, tt.args.reconcilerConfig)).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestLeaderElectionManager_start(t *testing.T) {
+	type fields struct {
+		workers                                []Worker
+		connectionFactory                      *db.ConnectionFactory
+		tearDown                               chan struct{}
+		leaderElectionReconcilerRepeatInterval time.Duration
+		workerGrp                              *sync.WaitGroup
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		setupFn func()
+	}{
+		{
+			name: "should start the worker if the worker is a leader and is not running",
```

Comment:
I attempted to add more test cases for the tests in this file  but every time I tried I kept running into issues with the sync.WaitGroup{}. It would constantly panic with the error `panic: sync: negative WaitGroup counter`. I was able to get around this by using .add(). But doing it this way broke the linter eg `copylocks: literal copies lock value from *tt.fields.workerGrp: sync.WaitGroup contains sync.noCopy`